src/components/FolderCreationModal.jsx

```diff
@@ -0,0 +1,237 @@
+import { Redirect } from 'react-router-dom';
+import update from 'immutability-helper';
+import Select from 'react-select';
+import FormField from './FormField';
+import FolderCard from './FolderCard';
+import LoadingButton from './LoadingButton';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import contentStyles from '../styles/isomer-cms/pages/Content.module.scss';
+import adminStyles from '../styles/isomer-cms/pages/Admin.module.scss';
+import { validateCategoryName } from '../utils/validators';
+import { toast } from 'react-toastify';
+import Toast from './Toast';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+const FolderCreationModal = ({
+  parentFolder,
+  existingSubfolders,
+  pagesData,
+  siteName,
+  setIsFolderCreationActive,
+}) => {
+  // Errors
+  const [errors, setErrors] = useState('')
+
+  const [folderName, setFolderName] = useState('');
+  const [title, setTitle] = useState('')
+  const [selectedFiles, setSelectedFiles] = useState(new Set())
+  //retrieve only when necessary, i.e. after user has chosen to move this page
+  // const [sha, setSha] = useState('')
+
+  const [isSelectingTitle, setIsSelectingTitle] = useState(true)
+  // Page redirection and modals
+  const [newPageUrl, setNewPageUrl] = useState('')
+  const [redirectToNewPage, setRedirectToNewPage] = useState(false)
+  const [sortedPagesData, setSortedPagesData] = useState(pagesData)
+
+  const sortOptions = [
+    ... parentFolder 
+      ? [{
+        value: 'collection',
+        label: 'Collection',
+      }]
+      : [],
+    {
+      value: 'title',
+      label: 'Name'
+    }
+  ]
+
+  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${parentFolder ? `/collections/${parentFolder}` : ''}`
+
+  const saveHandler = async () => {
+    // TODO
+    console.log('saving')
+  }
+
+  const folderNameChangeHandler = (event) => {
+    const { id, value } = event.target;
+    let errorMessage = validateCategoryName(value, 'page')
+    if (existingSubfolders.includes(value)) errorMessage = `Another folder with the same name exists. Please choose a different name.`
+    setTitle(value)
+    setErrors(errorMessage)
+  }
+
+  const fileSelectChangeHandler = (name) => {
+    let newSelectedFiles
+    if (selectedFiles.has(name)) {
+      newSelectedFiles = update(selectedFiles, {
+        $remove: [name]
+      })
+    } else {
+      newSelectedFiles = update(selectedFiles, {
+        $add: [name]
+      })
+    }
+    setSelectedFiles(newSelectedFiles)
+  }
+
+  const sortFuncs = {
+    'title': (a, b) => {
+      return a.title.localeCompare(b.title)
+    }
+  }
+
+  const sortOrderChangeHandler = (option) => {
+    // Reset to original order in collection
+    if (option.value === 'collection') setSortedPagesData(pagesData)
+    else {
+      const sortedOrder = sortedPagesData.concat().sort(sortFuncs[option.value])
+      setSortedPagesData(sortedOrder)
+    }
+  }
+
+  return (
```

Comment:
Discussed offline with Alex, some changes we could consider:

- Splitting up this FolderCreationModal into two files so that its more readable, 1) screen modal for folder name, which can be reused for editing folder name in future 2) screen for selecting files
- Letting redirection be handled in Folders.jsx rather than in FolderCreationModal.jsx, since both Folder creation and Page creation will require redirection.